### PR TITLE
Provide documentation

### PR DESCRIPTION
**Pull request by dhodovsk**
After merge, the documentation would be added to https://readthedocs.org/projects/conu/

To see the documentation now:
- run `make html` in docs directory
- view docs/build/html/index.html in your browser

WIP points
- [x] provide more examples and add cross-reference links to mentioned objects - partially done - more examples will appear when conu usage in real image will be implemented
- [x] figure out installation and document it in installation page
- [ ] ~~provide docstring for Volume class~~
- [x] add contribution page (?)

### CODE REVIEW

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/GPly](https://da.gd/GPly)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/GPly](https://da.gd/GPly)__
* Comment: __This is testing CI__

File: conu/apidefs/container.py [hunk 1]
```diff
@@ -194,12 +194,10 @@ def execute(self, command, **kwargs):
         """
         execute a command in this container -- usually the container needs to be running
 
-        TODO: what about parameters?
-
-        :param command: str, command to execute in the container
-        :param kwargs:
-        :return: ? we need to provide output, exit code and there needs to be a possibility for
-                  this thing to be async and blocking
```
Comment (TomasTomecek) [inline conu/apidefs/container.py]: created https://github.com/fedora-modularity/conu/issues/55 to track this work

File: conu/apidefs/container.py [hunk 2]
```diff
@@ -194,12 +194,10 @@ def execute(self, command, **kwargs):
...
+        :param command: list of str, command to execute in the container
+        :param exec_create_kwargs: dict, params to pass to exec_create()
```
Comment (TomasTomecek) [inline conu/apidefs/container.py]: these are not specified in the method definition -- could we list kwargs here and say those will container specific parameters for container engines?

File: docs/Makefile [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line.
+SPHINXOPTS    =
+SPHINXBUILD   = sphinx-build
+SPHINXPROJ    = Conu
```
Comment (TomasTomecek) [inline docs/Makefile]: I think we should organize a pub where we agree on how the project name is suppose to be spelled :)

File: docs/source/conf.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+#
+# source_suffix = ['.rst', '.md']
+source_suffix = '.rst'
+
+# The master toctree document.
+master_doc = 'index'
+
+# General information about the project.
+project = 'Conu'
+copyright = '2017, Userspace Containerisation'
```
Comment (TomasTomecek) [inline docs/source/conf.py]: should be Red Hat probably

File: docs/source/conf.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+author = 'Userspace Containerisation'
```
Comment (TomasTomecek) [inline docs/source/conf.py]: same here

File: docs/source/examples/s2i.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
+from conu.backend.docker.image import S2IDockerImage
```
Comment (TomasTomecek) [inline docs/source/examples/s2i.py]: Awesome!! I want these to be part of CI.

File: docs/source/installation.rst [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+.. code-block:: bash
+
+   sh -c conu/requirements.sh
+
+Add conu directory to PYTHONPATH variable, so it can be imported
+
+.. code-block:: bash
+
+   export PYTHONPATH=$PYTHONPATH:${PWD}/conu
```
Comment (TomasTomecek) [inline docs/source/installation.rst]: I installed it using `pip3 install --user .`, what do you think?

Comment (dhodovsk) [reply docs/source/installation.rst]: I think that it looks nicer. It should now be fixed.

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/Ufsk](https://da.gd/Ufsk)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/fF3If](https://da.gd/fF3If)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/apAj](https://da.gd/apAj)__
* Comment: __This is testing CI__

**Comment by TomasTomecek** [comment]
the examples are failing :/

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/Dcbwf](https://da.gd/Dcbwf)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/zUtU](https://da.gd/zUtU)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/qD247](https://da.gd/qD247)__
* Comment: __This is testing CI__

**Comment by TomasTomecek** [comment]
CI is green, merging; very nice work!

**Pull request merged**